components/sup/src/manager/mod.rs

```diff
@@ -826,6 +816,110 @@ impl Manager {
         ))
     }
 
+    pub fn service_cfg_validate(
+        mgr: &ManagerState,
+        req: &mut CtlRequest,
+        mut opts: protocols::ctl::SvcValidateCfg,
+    ) -> NetResult<()> {
+        if opts.get_cfg().len() > protocols::butterfly::MAX_SVC_CFG_SIZE {
+            return Err(net::err(
+                ErrCode::EntityTooLarge,
+                "Configuration too large.",
+            ));
+        }
+        if opts.get_format() != protocols::types::ServiceCfg_Format::TOML {
+            return Err(net::err(
+                ErrCode::NotSupported,
+                format!(
+                    "Configuration format {} not available.",
+                    opts.get_format()
+                ),
+            ));
+        }
+        let new_cfg = toml::from_slice(opts.get_cfg()).map_err(|e| {
+            net::err(
+                ErrCode::BadPayload,
+                format!(
+                    "Unable to decode configuration as {}, {}",
+                    opts.get_format(),
+                    e
+                ),
+            )
+        })?;
+        let service_group: ServiceGroup = opts.take_service_group().into();
+        for service in mgr.services.read().unwrap().iter() {
```

Comment:
What if the `unwrap` panics?

Reply:
Then the Supervisor is 💯 fucked already. It means the lock is poisoned and we don't want to handle it, we want to crash